go/libkb/remote_proof_links.go

```diff
@@ -139,25 +139,20 @@ func AddToProofSetNoChecks(r RemoteProofChainLink, ps *ProofSet) {
 
 func (r *RemoteProofLinks) active() []ProofLinkWithState {
 	var links []ProofLinkWithState
-	seen := make(map[string]bool)
+	seen := make(map[string]struct{})
 	for _, list := range r.links {
 		for i := len(list) - 1; i >= 0; i-- {
 			both := list[i]
 			link := both.link
-			if link.IsRevoked() {
-				continue
-			}
-
-			// We only want to use the last proof in the list
-			// if we have several (like for dns://chriscoyne.com)
-			id := link.ToDisplayString()
-			if seen[id] {
-				continue
+			if !link.IsRevoked() {
+				id := link.ToDisplayString()
+				// We only want to use the last proof in the list
+				// if we have several (like for dns://chriscoyne.com)
+				if _, ok := seen[id]; !ok {
+					links = append(links, both)
+					seen[id] = struct{}{}
```

Comment:
hmm, maybe not, good point. it's confusing though, since with HN at least, names are case-sensitive, but you can't have 2 usernames that are the same if toLower'ed.